pipelines/_common.toml

```diff
@@ -0,0 +1,62 @@
+# Commmon layout components for POPROX pipelines
+
+[meta]
+name = "NRMS"
+
+# Define convenient aliases for components so we do not need to re-type their
+# full names everywhere.  Inherting configurations can override these to change
+# the component without overwriting it!
+[definitions]
+# the name of the final personalized ranking component
+ranker = "ranker"
+# number of recommendation slots to configure
+N = 10
+
+[inputs.candidate]
+type = "poprox_concepts.domain.article.ArticleSet"
+[inputs.profile]
+type = "poprox_concepts.domain.article.ArticleSet"
+[inputs.clicked]
+type = "poprox_concepts.domain.article.InterestProfile"
+
+[components.candidate-embedder]
+code = "poprox_recommender.components.embedders.article:NRMSArticleEmbedder"
+config = { model_path = "nrms-mind/news_encoder.safetensors" }
+inputs = { article_set = "candidate" }
+
+[components.history-embedder]
+code = "poprox_recommender.components.embedders.article:NRMSArticleEmbedder"
+config = { model_path = "nrms-mind/news_encoder.safetensors" }
+inputs = { article_set = "candidate" }
+meta.group = "history-based ranking"
```

Comment:
This is only to allow an auto-generator of configuration graphs to group components together for better-looking charts.